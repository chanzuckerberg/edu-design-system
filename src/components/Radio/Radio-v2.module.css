/*------------------------------------*\
    # RADIO BUTTON
\*------------------------------------*/

/**
 * A custom individual radio control
 */
.radio {
  display: flex;
  gap: 0.5rem;
}

/**
 * Wraps the visually hidden radio input element and the visible sibling svg element.
 */
.input__wrapper {
  position: relative;
  /* Centers the radio icon in the wrapper. */
  display: inline-flex;
  align-items: center;
  /* Aligns the radio with the first line of the label. */
  align-self: flex-start;
}
/**
 * The visually hidden input element for the radio. The visual radio is provided by an svg element.
 */
.radio__input {
  /* Removes default margins placed by browser for radios. */
  margin: 0;
  /* Remove the radio from the page flow, positioning it on top of the SVG. */
  position: absolute;
  /* Set same dimensions as the RadioSvg element. */
  height: 1.5rem;
  width: 1.5rem;

  /**
   * Hide the input element visually.
   * This ensures the radio is still "physically" present so that all users,
   * especially on touch screen readers, still interact with the real radio element
   * where it would naturally be present.
   */
  opacity: 0;
}

/**
 * The disabled status of the visually hidden input element.
 */
 .radio__input:disabled {
  /* Needed since the input element overlays the visible svg icon for user input and cursor. */
  cursor: not-allowed;
  pointer-events: none;
}

.radio__labels {
  position: relative;
}

/**
 * Text that labels a radio input.
 */
.radio__label {
  position: relative;
}

.radio__sub-label {
  display: block;

  color: var(--eds-theme-color-text-utility-neutral-secondary);
}

/**
 * The visible radio svg icon element
 */
.radio__icon {
  /* Creates space for the border so that there's no jitter when the focus border is visible. */
  border: 0.125rem solid transparent;

  /* Theming when unchecked */
  .radio__input:not(:checked) + & {
    color: var(--eds-theme-color-border-utility-neutral-medium-emphasis);
  }

  .radio__input:not(:checked):hover + & {
    color: var(--eds-theme-color-border-utility-neutral-medium-emphasis-hover);
  }

  .radio__input:not(:checked):active + & {
    color: var(--eds-theme-color-border-utility-neutral-medium-emphasis-active);
  }

  .radio__input:not(:checked):disabled + & {
    color: var(--eds-theme-color-border-utility-disabled);
    pointer-events: none;
  }

  /* Theming when checked */
  .radio__input:checked + & {
    color: var(--eds-theme-color-background-utility-interactive-high-emphasis);
  }

  .radio__input:checked:hover ~ & {
    color: var(--eds-theme-color-background-utility-interactive-high-emphasis-hover);
  }

  .radio__input:checked:active ~ & {
    color: var(--eds-theme-color-background-utility-interactive-high-emphasis-active);
  }

  .radio__input:checked:disabled ~ & {
    color: var(--eds-theme-color-border-utility-disabled);
  }

  /**
   * Error Theming
   */ 
  .radio__input.radio--error ~ & {
    color: var(--eds-theme-color-border-utility-critical);
  }

  .radio__input.radio--error:hover ~ & {
    color: var(--eds-theme-color-border-utility-critical-hover);
  }

  .radio__input.radio--error:active ~ & {
    color: var(--eds-theme-color-border-utility-critical-active);
  }
}

/**
 * Handling focus ring
 * TODO-AH: focus ring clamped to the actual radio button, not its bounding label (browser default)
 */
.radio__input:focus-visible + .radio__icon {
  border: 0.125rem solid var(--eds-theme-color-border-utility-focus);
  border-radius: var(--eds-border-radius-full);
}

@supports not selector(:focus-visible) {
  .radio__input:focus + .radio__icon {
    border: 0.125rem solid var(--eds-theme-color-border-utility-focus);
    border-radius: var(--eds-border-radius-full);
  }
}
