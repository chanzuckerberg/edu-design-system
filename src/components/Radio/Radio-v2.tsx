import clsx from 'clsx';
import React from 'react';
import type { ReactNode } from 'react';

import { FieldLabelV2 as FieldLabel } from '../..';
import { useId } from '../../util/useId';
import type { EitherInclusive } from '../../util/utility-types';

import Text from '../Text';

import styles from './Radio-v2.module.css';

type RadioHTMLElementProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  'checked' | 'id' | 'name'
>;

type RadioInputProps = RadioHTMLElementProps & {
  /**
   * Whether checkbox is checked.
   */
  checked?: boolean;
  /**
   * Additional classnames passed in for styling.
   */
  className?: string;
  /**
   * Radio ID. Used to connect the input with a label for accessibility purposes.
   */
  id?: string;
  /**
   * Whether the radio button is in an error state
   */
  isError?: boolean;
};

type RadioProps = RadioInputProps & {
  // Component API
  /**
   * HTML id attribute. If not passed, this component
   * will generate an id to use for accessibility.
   */
  id?: string;
  /**
   * The field name to use in a form
   */
  name?: string;
  // Design API
  /**
   * Whether the radio button is in an error state
   */
  isError?: boolean;
  /**
   * Additional descriptive text below the primary label, adding additional detail
   */
  subLabel?: string;
} & EitherInclusive<
    {
      /**
       * Visible text label for the component.
       */
      label: ReactNode;
    },
    {
      /**
       * Aria-label to provide an accesible name for the text input if no visible label is provided.
       */
      'aria-label': string;
    }
  >;

/**
 * Radio input element, exported for greater flexibility.
 * You must provide an `id` prop and connect it to a visible label.
 */
const RadioInput = ({
  checked,
  className,
  disabled,
  isError,
  ...other
}: RadioInputProps) => {
  return (
    <span
      className={clsx(
        styles['input__wrapper'],
        disabled && styles['input__wrapper--disabled'],
      )}
    >
      <input
        checked={checked}
        className={clsx(
          className,
          isError && styles['radio--error'],
          styles['radio__input'],
        )}
        disabled={disabled}
        type="radio"
        {...other}
      />
      <span className={styles['radio__container']}></span>
      <span className={styles['radio__selected']}></span>
    </span>
  );
};

/**
 * `import {Radio} from "@chanzuckerberg/eds";`
 *
 * Radio control indicating if one item is selected or unselected from a set of other options. Uncontrolled by default, it can be used in place of a select field in form data.
 *
 * NOTE: This component requires `label` or `aria-label` prop
 */
export const Radio = ({
  className,
  disabled,
  label,
  id,
  isError = false,
  subLabel,
  ...other
}: RadioProps) => {
  const generatedId = useId();
  const radioId = id || generatedId;

  const componentClassName = clsx(
    styles['radio'],
    isError && styles['radio--error'],
    className,
  );

  return (
    <div className={componentClassName}>
      <RadioInput
        disabled={disabled}
        id={radioId}
        isError={isError}
        {...other}
      />
      <div className={styles['radio__labels']}>
        <FieldLabel disabled={disabled} htmlFor={radioId}>
          {label}
        </FieldLabel>
        {subLabel && (
          <Text
            as="span"
            className={styles['radio__sub-label']}
            preset="body-sm"
          >
            {subLabel}
          </Text>
        )}
      </div>
    </div>
  );
};

Radio.displayName = 'Radio';
RadioInput.displayName = 'Radio.Input';

Radio.Input = RadioInput;
Radio.Label = FieldLabel;
