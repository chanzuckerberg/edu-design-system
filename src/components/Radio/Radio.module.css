/*------------------------------------*\
    # RADIO BUTTON
\*------------------------------------*/

/**
 * A custom individual radio control
 */
.radio {
  display: flex;
  gap: var(--eds-size-1);
}
