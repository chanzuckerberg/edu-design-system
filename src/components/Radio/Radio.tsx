import clsx from 'clsx';
import React, { ChangeEventHandler } from 'react';
import styles from './Radio.module.css';

export interface Props {
  /**
   * HTML id of the helper text used to describe the component
   */
  ariaDescribedBy?: string;
  /**
   * Toggles whether or not the checkbox is checked or unchecked
   */
  checked?: boolean;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Disables the field and prevents editing the contents
   */
  disabled?: boolean;
  /**
   * HTML id for the component
   */
  id?: string;
  /**
   * HTML name attribute for the radio button
   */
  name?: string;
  /**
   * Function that fires when field value has changed
   */
  onChange?: ChangeEventHandler;
  /**
   * Toggles the form control's interactivity. When `readOnly` is set to `true`, the form control is not interactive
   */
  readOnly?: boolean;
  /**
   * Tabindex passed down from RadioFieldItem to add or remove focus capability
   */
  tabIndex?: number;
  /**
   * Inverted variation for dark backgrounds
   */
  inverted?: boolean;
  /**
   * A string representing the value of the radio button
   */
  value?: string | number;
}

/**
 * Primary UI component for user interaction
 */
export const Radio = ({
  ariaDescribedBy,
  id,
  name,
  value,
  className,
  checked,
  disabled,
  inverted,
  readOnly,
  onChange,
  tabIndex,
  ...other
}: Props) => {
  const componentClassName = clsx(
    styles['radio'],
    className,
    inverted && styles['radio--inverted'],
  );
  return (
    <div className={componentClassName} {...other}>
      <input
        aria-describedby={ariaDescribedBy}
        checked={checked}
        className={styles['radio__input']}
        disabled={disabled}
        id={id}
        name={name}
        onChange={onChange}
        readOnly={readOnly}
        tabIndex={tabIndex}
        type="radio"
        value={value}
      />
      <span className={styles['radio__custom-radio']} />
    </div>
  );
};
