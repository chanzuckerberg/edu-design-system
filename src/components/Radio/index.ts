export { Radio as default } from './Radio';
export { Radio as RadioV2 } from './Radio-v2';
