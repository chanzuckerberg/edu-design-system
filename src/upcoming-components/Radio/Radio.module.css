@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    # RADIO BUTTON
\*------------------------------------*/

/**
 * A custom individual radio control
 * 1) Outer wrapper div for custom control
 * 2) Determines width and height of the control
 * 3) Prevents control from being "squished" in 
 *    a flexbox parent
 */
.radio {
  position: relative;
  display: block;
  width: var(--eds-size-2-and-half); /* 2 */
  height: var(--eds-size-2-and-half); /* 2 */
  flex: none; /* 3 */
}

/**
 * Radio control input
 * 1) Make control dimensions same as outer container
 * 2) Hide native form controls
 */
.radio__input {
  position: absolute;
  top: 0; /* 1 */
  right: 0; /* 1 */
  bottom: 0; /* 1 */
  left: 0; /* 1 */
  z-index: var(--eds-z-index-400);
  display: block;
  width: 100%; /* 1 */
  height: 100%; /* 1 */
  cursor: pointer;
  opacity: 0; /* 2 */

  &:disabled {
    cursor: not-allowed;
  }
}

/**
 * Custom radio
 * 1) Make control dimensions same as outer container
 */
.radio__custom-radio {
  position: absolute;
  top: 0; /* 1 */
  right: 0; /* 1 */
  bottom: 0; /* 1 */
  left: 0; /* 1 */

  /**
     * Custom radio background 
     * 1) Make control dimensions same as outer container
     * 2) Stack background behind radio foreground
     */
  &:before {
    content: '';
    position: absolute;
    top: 0; /* 1 */
    right: 0; /* 1 */
    bottom: 0; /* 1 */
    left: 0; /* 1 */
    z-index: var(--eds-z-index-100); /* 2 */
    display: block;
    border-width: var(--eds-theme-form-border-width);
    border-style: solid;
    border-color: var(--eds-theme-color-form-input-border);
    border-radius: 50%;
    background: transparent;

    &:hover {
      border-color: var(--eds-theme-color-form-input-border-hover);
    }

    /**
     *  Custom radio background inside of focused control 
     */
    .radio__input:focus-visible + & {
      @mixin focus;
    }

    /**
     *  Custom radio background inside of disabled control 
     */
    .radio__input:disabled + & {
      border-color: var(--eds-theme-color-disabled-border);
      background-color: var(--eds-theme-color-background-disabled);
      cursor: not-allowed;
    }
  }

  /**
     * Custom radio inner circle
     * 1) Position circle in center of box
     * 2) Show/hide circle based on if input is checked
     */
  &:after {
    content: '';
    position: absolute;
    top: 6px; /* 1 */
    right: 6px; /* 1 */
    bottom: 6px; /* 1 */
    left: 6px; /* 1 */
    z-index: var(--eds-z-index-200);
    display: block;
    border: 4px solid var(--eds-theme-color-background-brand-primary-strong);
    border-radius: 50%;
    background: var(--eds-theme-color-background-brand-primary-strong);
    opacity: 0; /* 2 */

    .radio--inverted & {
      border-color: var(--eds-theme-color-border-brand-primary-subtle);
    }

    .radio__input:checked + & {
      opacity: 1; /* 2 */
    }
  }
}
