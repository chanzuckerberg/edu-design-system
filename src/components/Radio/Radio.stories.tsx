import {BADGE} from '@geometricpanda/storybook-addon-badges';
import {StoryObj, Meta} from '@storybook/react';
import React from 'react';

import {Radio} from './Radio';

export default {
  title: 'Atoms/Forms/Radio',
  component: Radio,
  args: {
    label: 'Option 1',
    checked: false,
  },
  parameters: {
    badges: [BADGE.BETA],
  },
  decorators: [
    (Story) => (
      <div style={{margin: '0.25rem'}}>
        <Story />
      </div>
    ),
  ],
} as Meta<Args>;

/**
 * Controlled example to make checked stories interactive.
 */
function RadioExample(args: Args) {
  const [checked, setChecked] = React.useState(args.checked);
  const handleChange = () => {
    setChecked(!checked);
  };

  return <Radio {...args} checked={checked} onChange={handleChange} />;
}

type Args = React.ComponentProps<typeof Radio>;

export const Default: StoryObj<Args> = {
  render: (args) => <RadioExample {...args} />,
};

export const Medium: StoryObj<Args> = {
  args: {
    size: 'md',
  },
  render: (args) => <RadioExample {...args} />,
};

export const Checked: StoryObj<Args> = {
  args: {
    checked: true,
  },
  render: (args) => <RadioExample {...args} />,
};

export const CheckedMedium: StoryObj<Args> = {
  args: {
    checked: true,
    size: 'md',
  },
  render: (args) => <RadioExample {...args} />,
};

export const Disabled: StoryObj<Args> = {
  args: {
    disabled: true,
  },
  render: (args) => <RadioExample {...args} />,
  parameters: {
    axe: {
      disabledRules: ['color-contrast'],
    },
  },
};

export const WithoutVisibleLabel: StoryObj<Args> = {
  render: () => (
    <>
      <Radio aria-label="unchecked radio" readOnly />
      <Radio aria-label="checked radio" checked readOnly />
      <Radio aria-label="unchecked disabled radio" disabled />
      <Radio aria-label="checked disabled radio" checked disabled />
    </>
  ),
  parameters: {
    axe: {
      disabledRules: ['color-contrast'],
    },
  },
};

export const LongLabels = {
  render: () => {
    const label = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit';

    return (
      <div
        style={{
          display: 'grid',
          width: '20rem',
          gridTemplateColumns: 'repeat(2, minmax(0, 1fr))',
          gap: '1rem',
        }}
      >
        <Radio label={label} readOnly />
        <Radio label={label} readOnly size="md" />
        <Radio disabled label={label} />
        <Radio disabled label={label} size="md" />
      </div>
    );
  },
  parameters: {
    axe: {
      disabledRules: ['color-contrast'],
    },
  },
};

export const WithCustomPositioning = {
  render: () => (
    <div style={{display: 'flex', alignItems: 'center'}}>
      <Radio.Label htmlFor="test">Label on Left</Radio.Label>
      <Radio.Input checked={false} id="test" readOnly />
    </div>
  ),
};
