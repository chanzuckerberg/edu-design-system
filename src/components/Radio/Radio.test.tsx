import { generateSnapshots } from '@chanzuckerberg/story-utils';
import { composeStories } from '@storybook/testing-react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Radio } from './Radio';
import * as stories from './Radio.stories';

const { Default } = composeStories(stories);

describe('<Radio />', () => {
  generateSnapshots(stories);
  test('throws an error if no label or aria-label', () => {
    expect(() => {
      render(<Radio />);
    }).toThrow(/must provide a visible label or aria-label/);
  });

  // eslint-disable-next-line jest/no-disabled-tests
  test.skip('should toggle the radio with space', async () => {
    const user = userEvent.setup();
    render(<Default />);
    const radio = screen.getByRole('radio');
    radio.focus();
    await user.keyboard(' ');
    expect(radio).toBeChecked();
  });
});
