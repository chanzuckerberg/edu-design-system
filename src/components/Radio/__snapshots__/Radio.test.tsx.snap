// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> Checked story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="radio"
  >
    <span
      class="input__wrapper"
    >
      <input
        checked=""
        class="radio__input"
        data-bootstrap-override="radio"
        id="5"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-selected"
        />
      </svg>
    </span>
    <label
      class="label label--lg radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="5"
    >
      Option 1
    </label>
  </div>
</div>
`;

exports[`<Radio /> CheckedMedium story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="radio"
  >
    <span
      class="input__wrapper"
    >
      <input
        checked=""
        class="radio__input"
        data-bootstrap-override="radio"
        id="7"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-selected"
        />
      </svg>
    </span>
    <label
      class="label label--md radio__label radio__label--md"
      data-bootstrap-override="label"
      for="7"
    >
      Option 1
    </label>
  </div>
</div>
`;

exports[`<Radio /> Default story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="radio"
  >
    <span
      class="input__wrapper"
    >
      <input
        class="radio__input"
        data-bootstrap-override="radio"
        id="1"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-unselected"
        />
      </svg>
    </span>
    <label
      class="label label--lg radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="1"
    >
      Option 1
    </label>
  </div>
</div>
`;

exports[`<Radio /> Disabled story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="radio"
  >
    <span
      class="input__wrapper input__wrapper--disabled"
    >
      <input
        class="radio__input"
        data-bootstrap-override="radio"
        disabled=""
        id="9"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon radio__icon--disabled"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-unselected"
        />
      </svg>
    </span>
    <label
      class="label label--lg label--disabled radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="9"
    >
      Option 1
    </label>
  </div>
</div>
`;

exports[`<Radio /> LongLabels story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    style="display: grid; width: 20rem; grid-template-columns: repeat(2, minmax(0, 1fr)); gap: 1rem;"
  >
    <div
      class="radio"
    >
      <span
        class="input__wrapper"
      >
        <input
          class="radio__input"
          data-bootstrap-override="radio"
          id="19"
          readonly=""
          type="radio"
        />
        <svg
          aria-hidden="true"
          class="icon radio__icon"
          fill="currentColor"
          height="1.625rem"
          style="--icon-size: 1.625rem;"
          width="1.625rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="test-file-stub#radio-unselected"
          />
        </svg>
      </span>
      <label
        class="label label--lg radio__label radio__label--lg"
        data-bootstrap-override="label"
        for="19"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </label>
    </div>
    <div
      class="radio"
    >
      <span
        class="input__wrapper"
      >
        <input
          class="radio__input"
          data-bootstrap-override="radio"
          id="21"
          readonly=""
          type="radio"
        />
        <svg
          aria-hidden="true"
          class="icon radio__icon"
          fill="currentColor"
          height="1.625rem"
          style="--icon-size: 1.625rem;"
          width="1.625rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="test-file-stub#radio-unselected"
          />
        </svg>
      </span>
      <label
        class="label label--md radio__label radio__label--md"
        data-bootstrap-override="label"
        for="21"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </label>
    </div>
    <div
      class="radio"
    >
      <span
        class="input__wrapper input__wrapper--disabled"
      >
        <input
          class="radio__input"
          data-bootstrap-override="radio"
          disabled=""
          id="23"
          type="radio"
        />
        <svg
          aria-hidden="true"
          class="icon radio__icon radio__icon--disabled"
          fill="currentColor"
          height="1.625rem"
          style="--icon-size: 1.625rem;"
          width="1.625rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="test-file-stub#radio-unselected"
          />
        </svg>
      </span>
      <label
        class="label label--lg label--disabled radio__label radio__label--lg"
        data-bootstrap-override="label"
        for="23"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </label>
    </div>
    <div
      class="radio"
    >
      <span
        class="input__wrapper input__wrapper--disabled"
      >
        <input
          class="radio__input"
          data-bootstrap-override="radio"
          disabled=""
          id="25"
          type="radio"
        />
        <svg
          aria-hidden="true"
          class="icon radio__icon radio__icon--disabled"
          fill="currentColor"
          height="1.625rem"
          style="--icon-size: 1.625rem;"
          width="1.625rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="test-file-stub#radio-unselected"
          />
        </svg>
      </span>
      <label
        class="label label--md label--disabled radio__label radio__label--md"
        data-bootstrap-override="label"
        for="25"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </label>
    </div>
  </div>
</div>
`;

exports[`<Radio /> Medium story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="radio"
  >
    <span
      class="input__wrapper"
    >
      <input
        class="radio__input"
        data-bootstrap-override="radio"
        id="3"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-unselected"
        />
      </svg>
    </span>
    <label
      class="label label--md radio__label radio__label--md"
      data-bootstrap-override="label"
      for="3"
    >
      Option 1
    </label>
  </div>
</div>
`;

exports[`<Radio /> WithCustomPositioning story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    style="display: flex; align-items: center;"
  >
    <label
      class="label label--lg radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="test"
    >
      Label on Left
    </label>
    <span
      class="input__wrapper"
    >
      <input
        class="radio__input"
        data-bootstrap-override="radio"
        readonly=""
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-unselected"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`<Radio /> WithoutVisibleLabel story renders snapshot 1`] = `
<div
  style="margin: 0.25rem;"
>
  <div
    class="radio"
  >
    <span
      class="input__wrapper"
    >
      <input
        aria-label="unchecked radio"
        class="radio__input"
        data-bootstrap-override="radio"
        id="11"
        readonly=""
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-unselected"
        />
      </svg>
    </span>
    <label
      class="label label--lg radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="11"
    />
  </div>
  <div
    class="radio"
  >
    <span
      class="input__wrapper"
    >
      <input
        aria-label="checked radio"
        checked=""
        class="radio__input"
        data-bootstrap-override="radio"
        id="13"
        readonly=""
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-selected"
        />
      </svg>
    </span>
    <label
      class="label label--lg radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="13"
    />
  </div>
  <div
    class="radio"
  >
    <span
      class="input__wrapper input__wrapper--disabled"
    >
      <input
        aria-label="unchecked disabled radio"
        class="radio__input"
        data-bootstrap-override="radio"
        disabled=""
        id="15"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon radio__icon--disabled"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-unselected"
        />
      </svg>
    </span>
    <label
      class="label label--lg label--disabled radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="15"
    />
  </div>
  <div
    class="radio"
  >
    <span
      class="input__wrapper input__wrapper--disabled"
    >
      <input
        aria-label="checked disabled radio"
        checked=""
        class="radio__input"
        data-bootstrap-override="radio"
        disabled=""
        id="17"
        type="radio"
      />
      <svg
        aria-hidden="true"
        class="icon radio__icon radio__icon--disabled"
        fill="currentColor"
        height="1.625rem"
        style="--icon-size: 1.625rem;"
        width="1.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="test-file-stub#radio-selected"
        />
      </svg>
    </span>
    <label
      class="label label--lg label--disabled radio__label radio__label--lg"
      data-bootstrap-override="label"
      for="17"
    />
  </div>
</div>
`;
